### DISCUSSION TITLE
How does the model handle Training

### DISCUSSION THREAD

**Discussion by dysruptllc** [facebookexperimental/Robyn#694] [Q&A]
We've reviewed a number of documents, but nothing we read explains explicitly how the model is handling training other than it's randomized as part of Nevergrad. 

Does anyone have a high-detail breakdown of how Robyn is handling this?

**Comment by CJ2407** [comment]
@laresbernardo I would like to follow-up on the above question with a clarifying question - does Robyn choose train, test, and val datasets randomly from the given modeling window? For e.g. my modeling input data has data from 2/1/2019 till 04/30/2022. So, Robyn chooses train data set sequentially from 2/1/2019 to let's say 11/1/2020, and then Val data from 11/2/2020 to 07/2/2021 and similarly for test. Or, it chooses these 3 data sets randomly? Clarifying cz of the vertical lines in this plot that suggests sequential splitting?
![image](https://user-images.githubusercontent.com/91983160/234332227-df9b647c-ce61-4839-a85a-ce78e6753d9f.png)

Also, is there a way to identify what data points in the time series became part of Train, Val, and Test looking at any of the model outputs?

**Comment by laresbernardo** [comment]
Taken from the comment in the demo:

> 
```
## Time series validation: When ts_validation = TRUE in robyn_run(), train_size defines the
# percentage of data used for training, validation and out-of-sample testing. For example,
# when train_size = 0.7, val_size and test_size will be 0.15 each. This hyperparameter is
# customizable with default range of c(0.5, 0.8) and must be between c(0.1, 1).
```

> Also, is there a way to identify what data points in the time series became part of Train, Val, and Test looking at any of the model outputs?

You can either calculate with the train_size value picked on the hyperparameters for your selected model (0.56 and 0.22 for validation and test sizes) or get it out of that specific plot in the one-pager too.

**Reply by CJ2407** [reply]
I think my question wasn't very clear - you see I have 3 years and 1 quarter of data from 2/1/2019 till 4/30/2022 which is 1185 days of data. I understand that by selecting train_size hyperparameter I am able to choose what % of 1185 rows of data will become my training data set, but my question is does Robyn choose number of rows for Train/Test/Val randomly out of 1185 from different year/month/weeks randomly or the sequentially meaning first 500 days of time series will become part of training data set and rest 500 are divided equally but sequentially into Validation and Test data set?

**Comment by laresbernardo** [comment]
We don't select those dates randomly but sequentially as shown in the plot. First part train, second part validation, third part test, unless `ts_validation` is set to FALSE, then we train and test with all data available from the modeling window.
